### PR TITLE
fix: broken links in docs

### LINKED ISSUE #785: Broken links in docs
There are some broken links that caught my attention, the details will be in the PR.

### PR DESCRIPTION
**Pull request by supiket**
- Changed `provider/external-node.md` links to `provider/connecting-to-an-external-node.md` links.
- Changed links in the following section of guide's welcome page:
  > See also the main The Fuel TypeScript SDK and Contributing pages for additional resources.
  
  Both the "The Fuel TypeScript SDK" and "Contributing" links were pointing to nowhere when I tried, if they should not be changed or should be changed to different links, I can update as needed.

Closes #785

### CODE REVIEW

File: docs/_guide/index.md [hunk 1]
```diff
@@ -14,4 +14,4 @@ This book is an overview of the different things one can achieve using the TypeS
 
 [The Fuel TypeScript SDK](./index.md)
```
Comment (Dhaiwat10) [inline docs/_guide/index.md:15]:
```suggestion
[The Fuel TypeScript SDK](../index.md)
```

File: docs/_guide/index.md [hunk 2]
```diff
@@ -14,4 +14,4 @@ This book is an overview of the different things one can achieve using the TypeS
...
 
-See also the main [The Fuel TypeScript SDK](./index.md) and [Contributing](../contributing.md) pages for additional resources.
+See also the main [The Fuel TypeScript SDK](https://github.com/FuelLabs/fuels-ts) and [Contributing](https://github.com/FuelLabs/fuels-ts/blob/master/CONTRIBUTING.md) pages for additional resources.
```
Comment (Dhaiwat10) [inline docs/_guide/index.md:17]:
```suggestion
See also the main [The Fuel TypeScript SDK](../index.md) and [Contributing](../../CONTRIBUTING.md) pages for additional resources.
```

File: docs/guide/index.md [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
 # The Fuel TypeScript SDK Guide
 
 TypeScript SDK for Fuel. It can be used for a variety of things, including but not limited to:
 
 - Compiling, deploying, and testing [Sway](https://github.com/FuelLabs/sway) contracts;
-- Use the [Testnet](../providers/external-node.md) or run a local Fuel node;
+- Use the [Testnet](./providers/connecting-to-an-external-node.md) or run a local Fuel node;
 - Crafting and signing transactions with hand-crafted scripts or contract calls;
 - Generating type-safe TypeScript bindings of contract ABI methods;
 - And more. `fuels-ts` is still in active development.
```
Comment (Dhaiwat10) [inline docs/guide/index.md:18]: We actually don't need to change any files under `docs/guide`. All the contents of this folder are auto-generated by our CI. We just need to change the files under `docs/_guide`, which you have already done. Please feel free to revert the changes you made to `docs/guide/index.md` and `docs/guide/providers/connection.md`!

**Review by Dhaiwat10 — CHANGES_REQUESTED** [review]
Good spot, thanks! Added some corrections for file paths and a comment or two 😄

**Review by arboleya — APPROVED** [review]

**Review by Dhaiwat10 — APPROVED** [review]

**Pull request merged**